.github/workflows/pull.yml

```diff
@@ -0,0 +1,37 @@
+# This workflow runs for every pull request
+name: PR CI
+
+on:
+  pull_request:
+
+jobs:
+  pr-ci-verification:
+    strategy:
+      matrix:
+        dafny-version: [
+          4.1.0,
+          4.3.0
```

Comment:
You talk about 4.2 in the PR but don't test it here?

Also, given the repetition of version you may want to pull that out as well.
Just saying

Reply:
Fair. Although getting to 4.3 across the board is more important than 4.2 in the broader context.

Reply:
I was actually a little worried about creating so many individual GHA jobs. As part of a different PR I'd like to first get the parameterization of test models out of workflow files: https://github.com/smithy-lang/smithy-dafny/pull/257/files#diff-a933719600e0ce4e5ada8a6a7a64dfb38e005f6145c00147d9a02a16c69409d7R20-R24 After that I'd be happier to test on more dafnies.